spring-cloud-sleuth-instrumentation/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/feign/SleuthFeignBuilder.java

```diff
@@ -36,16 +36,23 @@ private SleuthFeignBuilder() {
 	}
 
 	public static Feign.Builder builder(BeanFactory beanFactory) {
-		return Feign.builder().retryer(Retryer.NEVER_RETRY).client(client(beanFactory));
+		return builder(beanFactory, null);
 	}
 
-	private static Client client(BeanFactory beanFactory) {
-		try {
-			return new LazyClient(beanFactory);
-		}
-		catch (BeansException ex) {
-			return new LazyClient(beanFactory, new Client.Default(null, null));
+	public static Feign.Builder builder(BeanFactory beanFactory, Client delegate) {
+		return Feign.builder().retryer(Retryer.NEVER_RETRY).client(client(beanFactory, delegate));
+	}
+
+	private static Client client(BeanFactory beanFactory, Client delegate) {
+		if (delegate == null) {
+			try {
+				return new LazyClient(beanFactory);
+			}
+			catch (BeansException ex) {
```

Comment:
I know this was already there but I don't really get how this can throw an exception. `LazyClient` seems to handle this case already, I think we can remove from here.

Reply:
Removed.